stl/inc/yvals_core.h

```diff
@@ -1349,6 +1349,14 @@
 #endif // __cpp_impl_coroutine
 
 #if _HAS_CXX20

```

Comment:
After reading through this, I think it could be reasonable to implement "downlevel", just want to make sure this is intentional.

Reply:
It seems that in some rare cases we should not check concepts (see #2358) unless required by the standard. But I think it's OK to back port the changes to C++20 mode. We can defer the reconsideration until someone complains.

Reply:
It did seem to me like a DR; especially since it's strictly allowing more types in many of these algorithms, I don't feel uncomfortable backporting the changes. If others think it'd be better not to backport, I'm fine with that too, though.

Reply:
The typical reasons to implement a feature downlevel are:

* Invasively affects product code, such that supporting two modes would be a headache, and
* Doesn't disrupt user code that isn't taking advantage of the feature

This has to be balanced against the portability implications of implementing a feature downlevel (i.e. it allows code to be written with MSVC's STL that wouldn't work with libc++ or libstdc++, if they decide differently).

In general, anything that adds user-visible names is something that we guard for new language modes.

The `enable_shared_from_this` overhaul is a good example of something we implemented unconditionally.

Here, I am slightly concerned about the portability implications issue, but it doesn't seem onerous (only individual algorithm calls would be affected), and this is detectable via a feature-test macro *and* is specific to C++20 and above which is quite new (anything that goes back to C++14 would need more caution).

So I am ok with this, but I asked @CaseyCarter in the Discord just to be extra sure.

Reply:
I agree this is reasonable in C++20 mode since we're not adding new names, and (hopefully) not changing observable behavior for things that already compiled. We should probably poke the super secret "Standard Library Implementors" email list (which I've forgotten the name of) to let folks know we're implementing this in C++20.